modules/auth/ldap/ldap.go

```diff
@@ -64,7 +64,7 @@ func (ls *Source) sanitizedUserQuery(username string) (string, bool) {
 		return "", false
 	}
 
-	return fmt.Sprintf(ls.Filter, username), true
+	return strings.Replace(ls.Filter, "%s", username, -1), true
```

Comment:
Would it be possible to use the explicit argument indices supported by `fmt.Sprintf` (e.g. `"%[0]s, again I say %[0]s"`) instead of reinventing the wheel?

Reply:
The problem there was that my userDN didn't containt any string to format.  I needed to use a DN similar to this one: `cn=Users, dc=domain, dc=org` because I'm binding with userPrincipalName which is the one that needs the string format. Leaving the user DN untouched, I got an error, fmt.Sprintf returns `cn=Users, dc=domain, dc=org %!(EXTRA string=Username)` and that causes the search to fail.

As I mentioned above, I'm new to Go. If there's a better way to use the fmt without it failing if there's no string to format, I'll use it.